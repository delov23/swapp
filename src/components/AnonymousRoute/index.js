export { default as AnonymousRoute } from './AnonymousRoute';
