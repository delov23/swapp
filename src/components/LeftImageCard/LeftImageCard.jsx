import React, { useContext } from 'react';
import { Link } from 'react-router-dom';

import styles from './LeftImageCard.module.css';
import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';

const LeftImageCard = ({ width, image, text, href }) => {
  const theme = useContext(ThemeContext);

  return (
    <Link
      to={href}
      style={{ width }}
      className={styles[themify('card', theme)]}
    >
      <img className={styles.image} src={image} alt="" />
      <h3 className={styles.heading}>{text}</h3>
    </Link>
  );
};

export default LeftImageCard;
