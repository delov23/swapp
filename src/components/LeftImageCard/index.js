export { default as LeftImageCard } from './LeftImageCard';
