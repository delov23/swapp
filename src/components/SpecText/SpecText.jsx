import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import ThemeContext from '../../context/ThemeContext';
import { themify } from '../../utils/themify';
import styles from './SpecText.module.css';

const SpecText = ({ left, right }) => {
  const theme = useContext(ThemeContext);

  return (
    <p className={styles.text}>
      <span className={styles[themify('left', theme)]}>{left}: </span>
      <span className={styles[themify('right', theme)]}>{right}</span>
    </p>
  );
};

SpecText.propTypes = {
  left: PropTypes.string.isRequired,
  right: PropTypes.string.isRequired,
};

export default SpecText;
