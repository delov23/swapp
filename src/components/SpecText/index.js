export { default as SpecText } from './SpecText';
