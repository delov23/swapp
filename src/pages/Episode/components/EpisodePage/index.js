export { default as EpisodePage } from './EpisodePage';
