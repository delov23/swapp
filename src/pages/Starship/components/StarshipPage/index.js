export { default as StarshipPage } from './StarshipPage';
