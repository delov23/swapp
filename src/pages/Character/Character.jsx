import React from 'react';
import { CharacterPage } from './components/CharacterPage';

const Character = ({ match: { params } }) => {
  // const { characterId } = params;

  return <CharacterPage />;
};

export default Character;
