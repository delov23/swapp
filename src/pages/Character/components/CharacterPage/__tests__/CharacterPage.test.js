import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import { CharacterPage } from '../';
import { SpecText } from '../../../../../components/SpecText';
import { LeftImageCard } from '../../../../../components/LeftImageCard';
import { EXAMPLE_DATA } from '../../../__tests__/Character.test';

const EXAMPLE_PROPS = { ...EXAMPLE_DATA };

describe('<CharacterPage />', () => {
  it('should match snapshot', () => {
    expect(shallow(<CharacterPage {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should display basic information', () => {
    const wrapper = mount(
      <Router>
        <CharacterPage {...EXAMPLE_PROPS} />
      </Router>,
    );

    expect(wrapper.find(SpecText)).toHaveLength(4);
    expect(wrapper.find(LeftImageCard)).toHaveLength(2);
  });
});
