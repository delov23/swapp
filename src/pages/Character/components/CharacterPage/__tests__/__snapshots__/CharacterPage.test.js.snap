// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> should match snapshot 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": false,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": false,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": MockLink {
        "addTypename": false,
        "mockedResponsesByKey": Object {
          "{\\"query\\":\\"query character($id: ID!) {\\\\n  person(id: $id) {\\\\n    id\\\\n    name\\\\n    height\\\\n    mass\\\\n    image\\\\n    species {\\\\n      name\\\\n    }\\\\n    homeworld {\\\\n      name\\\\n    }\\\\n    starships {\\\\n      edges {\\\\n        node {\\\\n          id\\\\n          name\\\\n          image\\\\n        }\\\\n      }\\\\n    }\\\\n  }\\\\n}\\\\n\\"}": Array [
            Object {
              "request": Object {
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "character",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "arguments": Array [
                              Object {
                                "kind": "Argument",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "value": Object {
                                  "kind": "Variable",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                },
                              },
                            ],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "person",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "name",
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "height",
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "mass",
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "image",
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "species",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "name",
                                        },
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "homeworld",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "name",
                                        },
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "starships",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "edges",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "node",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "id",
                                                    },
                                                  },
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "name",
                                                    },
                                                  },
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "image",
                                                    },
                                                  },
                                                ],
                                              },
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [
                        Object {
                          "directives": Array [],
                          "kind": "VariableDefinition",
                          "type": Object {
                            "kind": "NonNullType",
                            "type": Object {
                              "kind": "NamedType",
                              "name": Object {
                                "kind": "Name",
                                "value": "ID",
                              },
                            },
                          },
                          "variable": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                        },
                      ],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 326,
                    "source": Object {
                      "body": "
  query character($id: ID!) {
    person(id: $id) {
      id
      name
      height
      mass
      image
      species {
        name
      }
      homeworld {
        name
      }
      starships {
        edges {
          node {
            id
            name
            image
          }
        }
      }
    }
  }
",
                      "locationOffset": Object {
                        "column": 1,
                        "line": 1,
                      },
                      "name": "GraphQL request",
                    },
                    "start": 0,
                  },
                },
                "variables": Object {
                  "id": "people.1",
                },
              },
              "result": [Function],
            },
          ],
        },
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {
            "{\\"query\\":\\"query character($id: ID!) {\\\\n  person(id: $id) {\\\\n    id\\\\n    name\\\\n    height\\\\n    mass\\\\n    image\\\\n    species {\\\\n      name\\\\n    }\\\\n    homeworld {\\\\n      name\\\\n    }\\\\n    starships {\\\\n      edges {\\\\n        node {\\\\n          id\\\\n          name\\\\n          image\\\\n        }\\\\n      }\\\\n    }\\\\n  }\\\\n}\\\\n\\"}": Array [
              Object {
                "request": Object {
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": Object {
                          "kind": "Name",
                          "value": "character",
                        },
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "arguments": Array [
                                Object {
                                  "kind": "Argument",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                },
                              ],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "person",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "height",
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "mass",
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "image",
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "species",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "name",
                                          },
                                        },
                                      ],
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "homeworld",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "name",
                                          },
                                        },
                                      ],
                                    },
                                  },
                                  Object {
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "starships",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "edges",
                                          },
                                          "selectionSet": Object {
                                            "kind": "SelectionSet",
                                            "selections": Array [
                                              Object {
                                                "arguments": Array [],
                                                "directives": Array [],
                                                "kind": "Field",
                                                "name": Object {
                                                  "kind": "Name",
                                                  "value": "node",
                                                },
                                                "selectionSet": Object {
                                                  "kind": "SelectionSet",
                                                  "selections": Array [
                                                    Object {
                                                      "arguments": Array [],
                                                      "directives": Array [],
                                                      "kind": "Field",
                                                      "name": Object {
                                                        "kind": "Name",
                                                        "value": "id",
                                                      },
                                                    },
                                                    Object {
                                                      "arguments": Array [],
                                                      "directives": Array [],
                                                      "kind": "Field",
                                                      "name": Object {
                                                        "kind": "Name",
                                                        "value": "name",
                                                      },
                                                    },
                                                    Object {
                                                      "arguments": Array [],
                                                      "directives": Array [],
                                                      "kind": "Field",
                                                      "name": Object {
                                                        "kind": "Name",
                                                        "value": "image",
                                                      },
                                                    },
                                                  ],
                                                },
                                              },
                                            ],
                                          },
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [
                          Object {
                            "directives": Array [],
                            "kind": "VariableDefinition",
                            "type": Object {
                              "kind": "NonNullType",
                              "type": Object {
                                "kind": "NamedType",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "ID",
                                },
                              },
                            },
                            "variable": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                          },
                        ],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 326,
                      "source": Object {
                        "body": "
  query character($id: ID!) {
    person(id: $id) {
      id
      name
      height
      mass
      image
      species {
        name
      }
      homeworld {
        name
      }
      starships {
        edges {
          node {
            id
            name
            image
          }
        }
      }
    }
  }
",
                        "locationOffset": Object {
                          "column": 1,
                          "line": 1,
                        },
                        "name": "GraphQL request",
                      },
                      "start": 0,
                    },
                  },
                  "variables": Object {
                    "id": "people.1",
                  },
                },
                "result": [Function],
              },
            ],
          },
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.3",
      "watchQuery": [Function],
    }
  }
>
  <Character
    match={
      Object {
        "params": Object {
          "characterId": 10,
        },
      }
    }
  />
</ApolloProvider>
`;

exports[`<CharacterPage /> should match snapshot 1`] = `
<main>
  <h1
    className="nameLight"
  >
    Name
  </h1>
  <hr />
  <section
    className="content"
  >
    <div
      className="leftBlock"
    >
      <Card
        image="https://image.com/img.jpg"
        title="Name"
      >
        <SpecText
          left="Height"
          right={200}
        />
        <SpecText
          left="Weight"
          right={209}
        />
        <SpecText
          left="Species"
          right="Spec"
        />
        <SpecText
          left="Home World"
          right="Home"
        />
      </Card>
    </div>
    <div
      className="rightBlock"
    >
      <h2
        className="headingLight"
      >
        Piloted Starships
      </h2>
      <hr />
      <LeftImageCard
        href="/starships/starship.1"
        image="https://image.com/img2.jpg"
        key="starship.1"
        text="Edge 1"
        width="100%"
      />
      <LeftImageCard
        href="/starships/starship.2"
        image="https://image.com/img3.jpg"
        key="starship.2"
        text="Edge 2"
        width="100%"
      />
    </div>
  </section>
</main>
`;
